lib/ld-eventsource/impl/event_parser.rb

```diff
@@ -65,7 +66,10 @@ def process_field(name, value)
             @data << value
             @have_data = true
           when "id"
-            @id = value
+            if !value.include?("\x00")
+              @id = value
+              @last_event_id = value
+            end
```

Comment:
This was another detail of the spec we weren't following. Event IDs are not allowed to contain a null character, because they might be sent in a Last-Event-Id header and HTTP header values can't contain nulls. We haven't been checking this in the SSE contract tests but probably should.